editor/plugins/org.fusesource.ide.projecttemplates/src/org/fusesource/ide/projecttemplates/util/BasicProjectCreatorRunnable.java

```diff
@@ -80,14 +83,9 @@ public void run(IProgressMonitor monitor) throws InvocationTargetException, Inte
 			IProject prj = c.getProject();
 			
 			if (ok) {
-				// then configure the project for the given template
-				AbstractProjectTemplate template = retrieveTemplate();
-				// now execute the template
-				try {
-					template.create(prj, metadata, subMonitor.split(1));
-				} catch (CoreException ex) {
-					ProjectTemplatesActivator.pluginLog().logError("Unable to create project...", ex); //$NON-NLS-1$
-				}
+				Display.getDefault().asyncExec( () ->  {
```

Comment:
I don't understand why it is done in an async as it is blocking for it anyways just the line after.

Reply:
I did that just to process UI events until the logic completes which should help keeping progressbar active and prevent "Application frozen" dialogs.

Reply:
updating the monitor is supposed to do this work